examples/example.lua

```diff
@@ -52,6 +52,16 @@ else
 	assert(example_input == decompress_deflate)
 end
 
+-- Compress asynchronously
+local compress_resume, compress_complete = LibDeflate:CompressDeflate(example_input, {level=9, async=true})
+repeat until not compress_resume()
+local compress_async = compress_complete()
+
+-- Decompress asynchronously
+local decompress_resume, decompress_complete = LibDeflate:DecompressDeflate(compress_deflate, {async=true})
+repeat until not decompress_resume()
+local decompress_async = decompress_complete()
+assert(decompress_async == example_input)
 
```

Comment:
Don't think two functions need to be returned. One function should be enough.
Note: Do not start (de)compression on direct calls to (De)CompressDeflate in async mode